Reference/Routing/Request-Pipeline/IContentFinder.md

```diff
@@ -91,6 +91,33 @@ this example shows how to add custom content finder to (and how to remove Conten
 
 #NotFoundHandlers
 
-To set your own 404 finder create an IContentFinder and set it as the ContentLastChanceFinder.  A ContentLastChanceFinder will always return a 404 status code. Example:
+To set your own 404 finder create an IContentFinder and set it as the ContentLastChanceFinder. A ContentLastChanceFinder will always return a 404 status code. This example creates a new implementation of the IContentFinder and checks whether the requested content could not be found by using the default `Is404` property presented in the `PublishedContentRequest` class.
+
+    public class My404ContentFinder : IContentFinder {
+    	public bool TryFindContent(PublishedContentRequest contentRequest) {
+            //logic to find your 404 page and set it to contentRequest.PublishedContent
+	     CultureInfo culture = null;
+            if (contentRequest.HasDomain) {
+                culture = CultureInfo.GetCultureInfo(contentRequest.UmbracoDomain.LanguageIsoCode);
+            }
+
+            //replace 'home_doctype_alias' with the alias of your homepage
+            IPublishedContent rootNode = UmbracoContext.Current.ContentCache.GetByXPath("//home_doctype_alias").FirstOrDefault(n => n.GetCulture().ThreeLetterWindowsLanguageName == culture.ThreeLetterWindowsLanguageName);
+            //replace '404_doctype_alias' with the alias of your 404 page
+            IPublishedContent notFoundNode = UmbracoContext.Current.ContentCache.GetByXPath("//404_doctype_alias").FirstOrDefault(n => n.GetCulture().ThreeLetterWindowsLanguageName == culture.ThreeLetterWindowsLanguageName);
+
+            if (notFoundNode != null) {
+                contentRequest.PublishedContent = notFoundNode;
+            } else if (rootNode != null) {
+                contentRequest.PublishedContent = rootNode;
+            } else {
+                contentRequest.PublishedContent = UmbracoContext.Current.ContentCache.GetAtRoot().FirstOrDefault(n => n.GetTemplateAlias() != "");
+            }
+
+            return contentRequest.PublishedContent != null;
```

Comment:
We should always return something here because it's the notFoundHandler.  We could also replace the code by a comment suggesting the user they should always have something to return.

Reply:
Hi Dampee, I've updated the code according to your input though i'm not sure about the returned result. I thought by checking if a node was assigned to the `contentRequest.PublishedContent ` then it's ok. Or do you mean to always assign a node that actually has a template and can be viewed like a page? Thanks!

Reply:
Yes, that is what I meant.